cli/src/main/java/com/devonfw/tools/ide/environment/EnvironmentVariablesPropertiesFile.java

```diff
@@ -104,27 +104,34 @@ public void save() {
       this.context.info("Converting legacy properties to {}", newPropertiesFilePath);
     }
     List<VariableLine> lines = new ArrayList<>();
-    // TODO FixMe: we should simply skip reading if not exists instead of creating an empty file and then reading it. 
-    //             Also we should do mkdirs on parent folder to ensure it exists - see Review from https://github.com/devonfw/IDEasy/pull/374
-    if (!Files.exists(newPropertiesFilePath)) {
-      try {
-        Files.createFile(newPropertiesFilePath);
-      } catch (IOException e) {
-        throw new IllegalStateException("Failed to create properties file with" + newPropertiesFilePath, e);
+
+    // Ensure the parent directory exists
+    try {
+      if (newPropertiesFilePath.getParent() != null) {
+        Files.createDirectories(newPropertiesFilePath.getParent());
       }
-    }
-    try (BufferedReader reader = Files.newBufferedReader(newPropertiesFilePath)) {
-      String line;
-      do {
-        line = reader.readLine();
-        if (line != null) {
-          VariableLine variableLine = VariableLine.of(line, this.context, getSource());
-          lines.add(variableLine);
-        }
-      } while (line != null);
     } catch (IOException e) {
-      throw new IllegalStateException("Failed to load existing properties from " + this.propertiesFilePath, e);
+      throw new IllegalStateException("Failed to create parent directories for " + newPropertiesFilePath, e);
+    }
+
+    // Skip reading if the file does not exist
+    if (Files.exists(newPropertiesFilePath)) {
+      try (BufferedReader reader = Files.newBufferedReader(newPropertiesFilePath)) {
+        String line;
+        do {
+          line = reader.readLine();
+          if (line != null) {
+            VariableLine variableLine = VariableLine.of(line, this.context, getSource());
+            lines.add(variableLine);
+          }
+        } while (line != null);
+      } catch (IOException e) {
+        throw new IllegalStateException("Failed to load existing properties from " + this.propertiesFilePath, e);
+      }
+    } else {
+      this.context.info("Properties file {} does not exist, skipping read.", newPropertiesFilePath);
```

Comment:
If we `save` but have nothing to read, DEBUG should be sufficient here to avoid spam.
```suggestion
      this.context.debug("Properties file {} does not exist, skipping read.", newPropertiesFilePath);
```